pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
+	ap := strings.Split(path, ".")
+	n := []map[string]interface{}{}
+	for _, v := range ap {
+		nm := make(map[string]interface{})
+		nm[v] = make(map[string]interface{})
+		n = append(n, nm)
+	}
+	for i, d := range n {
+		for k := range d {
+			z := i + 1
+			if z == len(n) {
+				n[i][k] = data
+				break
+			}
+			n[i][k] = n[z]
+		}
+	}
+
+	return n[0]
+}
+
+// getParents returns a slice of parent charts in reverse order
+func getParents(c *chart.Chart, out []*chart.Chart) []*chart.Chart {
+	if len(out) == 0 {
+		out = []*chart.Chart{}
+		out = append(out, c)
+	}
+	for _, ch := range c.Dependencies {
+		if len(ch.Dependencies) > 0 {
+			out = append(out, ch)
+			out = getParents(ch, out)
+		}
+	}
+
+	return out
+}
+
+// processImportValues merges values from child to parent based on ImportValues field
+func processImportValues(c *chart.Chart, v *chart.Config) error {
+	reqs, err := LoadRequirements(c)
+	if err != nil {
+		log.Printf("Warning: ImportValues cannot load requirements for %s", c.Metadata.Name)
+		return nil
+	}
+	cvals, err := CoalesceValues(c, v)
+	if err != nil {
+		log.Fatalf("Error coalescing values for ImportValues %s", err)
+	}
+	nv := v.GetValues()
+	b := make(map[string]interface{})
+	for kk, v3 := range nv {
+		b[kk] = v3
+	}
+	for _, r := range reqs.Dependencies {
+		if len(r.ImportValues) > 0 {
+			var outiv []interface{}
+			for _, riv := range r.ImportValues {
+				switch riv.(type) {
+				case map[string]interface{}:
+					if m, ok := riv.(map[string]interface{}); ok {
+						nm := make(map[string]string)
+						nm["child"] = m["child"].(string)
+						nm["parent"] = m["parent"].(string)
+						outiv = append(outiv, nm)
+						s := r.Name + "." + nm["child"]
+						vv, err := cvals.Table(s)
+						if err != nil {
+							log.Printf("Warning: ImportValues missing table %v", err)
+							continue
+						}
+						if nm["parent"] == "." {
+							coalesceTables(b, vv.AsMap())
+						} else {
+
+							vm := pathToMap(nm["parent"], vv.AsMap())
+							coalesceTables(b, vm)
+						}
+					}
+				case string:
+					nm := make(map[string]string)
+					nm["child"] = riv.(string)
```

Comment:
I perceive a problem or inconvenience with simple list form. I don't see it as a deal breaker but you cannot just use simple list to go from child->parent->grandparent because the imported value comes in at "." not in "exports." so the simple list import in grandparent cannot access parents values imported from child (with simple list). Just FYI in case you had plans to do child->parent->grandparent with just simple list.